src/v/crypto/README.md

```diff
@@ -12,3 +12,35 @@ This library provides the following implementations:
 * Cryptographically secure DRBG
 * Base64 decoding
 
+## Cryptographic Secure Digest
+
+There are effectively two ways of performing a cryptographically secure digest:
+
+* One-shot
+* Multi-part
+
+For a one-shot, operation you can simply do the following:
+
+```c++
+#include "crypto/crypto.h"
+
+ss::sstring msg = "Hi there, Redpanda!";
+
+auto digest = crypto::digest(
+  crypto::digest_type::SHA256,
+  {reinterpret_cast<bytes::value_type*>(msg.data()), msg.size()});
```

Comment:
I think we should aim for not recommending and requiring users of the library use `reinterpret_cast`.

Reply:
then I would need two methods of nearly the same signature, one for signed and one for unsigned buffers... do you have a recommendation for what that would look like?